log4j-api/src/main/java/org/apache/logging/log4j/util/Constants.java

```diff
@@ -107,7 +107,7 @@ private static int getMajorVersion() {
     }
 
     static int getMajorVersion(final String version) {
-        final String[] parts = version.split("-|\\.");
+        final String[] parts = version.split("-|\\.", -1);
```

Comment:
We are only interested in first two parts, shouldn't this be `2` instead?

Reply:
Thanks, I corrected it.
(actually we need 3 parts: `major.minor.rest_of_string`).